common/lib/xmodule/xmodule/modulestore/__init__.py

```diff
@@ -783,6 +785,34 @@ def ensure_indexes(self):
         """
         pass
 
+    @abstractmethod
+    def get_course_structure(self, course_id, version=None):
+        """
+        Returns a dict representing the course structure.
+
+        Args:
+            course_id (CourseKey):
+            version (string): some modulestores hold different versions of courses. A specific version
+                can be specified; otherwise, the latest published version will be returned.
+
+        The returned dict has the following format:
+        {
+            u'root': u'block-v1:edX+DemoX+2014_T1+type@course+block@course',
+            u'blocks': {
+                u'block-v1:edX+DemoX+2014_T1+type@sequential+block@basic_questions': {
+                u'block_type': u'sequential',
+               u'children': [u'block-v1:edX+DemoX+2014_T1+type@vertical+block@2152d4a4aadc4cb0af5256394a3d1fc7',
+                             u'block-v1:edX+DemoX+2014_T1+type@vertical+block@47dbd5f836544e61877a483c0b75606c',
+                             ...],
+               u'display_name': u'Homework - Question Styles',
+               u'format': u'Homework',
+               u'graded': True},
+               ...
+            }
+        }
+        """
+        pass
+
```

Comment:
You've documented the "structure" of `structure` well above - that puts this PR in front of lots of platform code that slings arounds dictionaries with reckless abandon. (What keys [are supposed to | should never] exist? Who the heck knows!) Have you considered an approach that returned a CourseStructure object with a set of fixed attributes?